dev/src/transaction.ts

```diff
@@ -256,10 +268,18 @@ export class Transaction {
    */
   set<T>(
     documentRef: DocumentReference<T>,
-    data: T,
+    data: T | Partial<T>,
     options?: SetOptions
   ): Transaction {
-    this._writeBatch.set(documentRef, data, options);
+    if (
+      options &&
+      (options.merge || options.mergeFields) &&
+      !isDefaultConverter(documentRef._converter)
+    ) {
+      this._writeBatch.set(documentRef, data as Partial<T>, options);
+    } else {
+      this._writeBatch.set(documentRef, data as T, options);
```

Comment:
Same as above - you should be able to remove this.

Reply:
done.